src/graph/graphcontext.c

```diff
@@ -250,24 +250,35 @@ uint GraphContext_AttributeCount(GraphContext *gc) {
 }
 
 Attribute_ID GraphContext_FindOrAddAttribute(GraphContext *gc, const char *attribute) {
+	// If we are already in a critical region, no further locking should be done in this function.
+	bool in_critical_region = QueryCtx_IsLocked();
```

Comment:
Can't we simply try to reacquire the lock, `Graph_AcquireReadLock`
basically changing the lock to be reentrant.
if we'll do that this function and check can be removed.